### PR TITLE
Small credential service refactoring

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: 
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: 
Small credential service refactoring for better performance. 
Before this change, anytime we initialize an HTTP source, we'd discover MSBuild which is costly and unnecessary.

## Testing/Validation
Tests Added: Yes  
Reason for not adding tests:  
Validation done:

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs [hunk 1]
```diff
@@ -21,21 +21,21 @@ public HttpHandlerResourceV3Provider()
         {
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs:21]: This file is basically reverted to the dev version

File: src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs [hunk 2]
```diff
@@ -3,7 +3,6 @@
 
 using System;
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs:4]: This file is basically reverted to the dev version

File: src/NuGet.Clients/NuGet.Credentials/CredentialService.cs [hunk 3]
```diff
@@ -34,20 +34,19 @@ public class CredentialService : ICredentialService
...
+        public bool HandlesDefaultCredentials { get; }
 
         /// <summary>
         /// Constructor
         /// </summary>
         /// <param name="providers">All available credential providers.</param>
         /// <param name="nonInteractive">If true, the nonInteractive flag will be passed to providers.
         /// NonInteractive requests must not promt the user for credentials.</param>
-        public CredentialService(AsyncLazy<IEnumerable<ICredentialProvider>> providers, bool nonInteractive)
+        public CredentialService(AsyncLazy<IEnumerable<ICredentialProvider>> providers, bool nonInteractive, bool handlesDefaultCredentials)
```
Comment (dtivel) [inline src/NuGet.Clients/NuGet.Credentials/CredentialService.cs]: Update XML doc comments.

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoveryUtility.cs [hunk 4]
```diff
@@ -7,11 +8,11 @@ namespace NuGet.Protocol.Plugins
 {
     public static class PluginDiscoveryUtility
     {
-        public static string InternalPluginDiscoveryRoot { get; set; }
+        public static Lazy<string> InternalPluginDiscoveryRoot { get; set; }
 
         public static string GetInternalPlugins()
         {
-            var rootDirectory = InternalPluginDiscoveryRoot ?? System.Reflection.Assembly.GetEntryAssembly()?.Location;;
+            var rootDirectory = InternalPluginDiscoveryRoot.Value ?? System.Reflection.Assembly.GetEntryAssembly()?.Location;;
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoveryUtility.cs]: Remove extra `;`.

**Review by dtivel — APPROVED** [review]

**Pull request merged**